web/access_control.go

```diff
@@ -78,6 +78,54 @@ func (s *AutograderService) isEnrolled(userID, courseID uint64) bool {
 	})
 }
 
+// isValidSubmission returns true if submitter has active course enrollment or
+// if submitting group belongs to the given course
+func (s *AutograderService) isValidSubmissionRequest(submission *pb.SubmissionRequest) bool {
+	if !submission.IsValid() {
+		fmt.Println("isValidSubmissionRequest failed: submission request not valid: ", submission)
+		return false
+	}
+	// ensure that group belongs to course
+	if submission.GetGroupID() > 0 {
+		group, err := s.db.GetGroup(submission.GetGroupID())
+		if err != nil || group.GetCourseID() != submission.GetCourseID() {
+			fmt.Println("isValidSubmissionRequest failed: could not get group ")
+			return false
+		}
+		return true
+	}
+	// ensure that user is enrolled in the course
+	if !s.hasCourseAccess(submission.GetUserID(), submission.GetCourseID(), func(e *pb.Enrollment) bool {
+		return e.Status >= pb.Enrollment_STUDENT
+	}) {
+		fmt.Println("isValidSubmissionRequest failed: failed hasCourseAccess ")
```

Comment:
Unless you want to log this, the if condition can be returned directly from this method without the if and brackets.

Reply:
Fixed.